docs/requirements.txt

```diff
@@ -1 +1,5 @@
 rst.linker
+mock
```

Comment:
Yeah, the code is unclear about what's the appropriate way to declare dependencies, in part because it's straddling two techniques (one which specifies the dependencies in `docs/requirements.txt` and another where they're specified in `.[doc]`). I dislike the `.[doc]` technique mainly because it requires that you install `.` to get doc dependencies, which isn't always necessary or desirable. I like the `.[doc]` technique because it keeps the dependencies declared in a single location (setup.py). I should point out that the `-e .` line also runs in conflict with the RTD option to install the package in a virtualenv, though presumably pip resolves that conflict in some reasonable way. In any case, it's messy and unclear in the original code, so I won't demand that someone trying to help with docs reconcile the mess.